### DISCUSSION TITLE
[일지] 5월 31일 (월) 오후 회의록

### DISCUSSION THREAD

**Discussion by olenmg** [bcaitech1/p4-opt-5-vibrhanium-#27] [일지]
### 스페셜 미션 3 
- 겹치는 파라미터를 따로 빼서 관리하면 더 간결
- 이미지 사이즈도 파라미터로 추가해서 통제하면 좋겠다.

### 스페셜 미션 4
- 데이터 줄이는 코드 
    - 비율을 어떻게 설정할 것인가?

### 코드 방향성
- model.yaml, 하이퍼파라미터.yaml => 모델 재현이 가능
- 그래프 찍어서 -> 성능 좋은거 재현
- 실험 및 시각화 -> wandb, db
- 모델을 쌓는 방법을 크게 다르게 하여 각자 실험

- 가능하다면, 최적 모델 찾고 -> 프루닝, 양자화

- 실험 환경 순서 구축
    1) 최적 모델 찾기 (하이퍼파라미터 고정시키고 찾아도 됨)
    2) 2-1) 최적모델의 하이퍼파라미터를 찾거나(더 딥하게)
    3) 2-2) pruning/(quantization)/distilation/tensor decomposition 적용

- quantization => batch size를 늘릴 수 있어 성능이 좋아질 수 있음
- wandb 연동

- (선순위) tensor decomposition을 잘 활용하기

- (후순위) 
- data imbalance를 해소 시킬 수 있는 loss 잘 설정하면 좋은 결과가 있을 것이다.
- 입력 이미지 크기가 영향을 많이 준다.
adjust loss : https://arxiv.org/pdf/2007.07314.pdf
- stage 1때 1등이 썼던 loss: Focal loss + F1 loss
    - loss를 혼합해서 쓰면 결과가 더 좋았다

- F1용으로 sample data로 테스트(재경 코드)
    
### 데일리 미션 나오고 있는 것
내일 : Week 2 Special Mission
변경 전: 6/3 (2주차 목) -> 변경 후: 6/1 (2주차 화)

### 그래서 어떤걸 할지
1. AutoML(optuna) 세팅 PR 넣기 
2. WandB 세팅
3. 일부 하이퍼파라미터 고정해서 최적 단일 모델 찾기
    - 고정할 것
        - AdamW optimizer
        - Warmup lr scheduler
        - CrossEntropyLoss criterion
        - Batch size, epoch, learning rate
        - input image size [28, 56, 112, 168, 224]
    - 변화시킬 것
        - 모델 구조
    - 모델 구조 변화시켜서 실험할 때 노션에 세팅 올려두기(+설명까지)

### 로봇 조아용
- [링크 1](https://smartstore.naver.com/robotnmore/products/5171555047)
- [링크 2](https://ddangeun.tistory.com/24?category=835029)

### 면접 질문(명환)
- 코테 3번 설명해보세요
시간복잡도 말해보세요

- 자기소개 해보세요

- 모델을 만드는과정을 설명해보세요
    - 전처리는?
    - 앙상블할때 중간에 나오는 모델을 다쓰는가
        - 앙상블의 종류에 대해서 말해보세요

- 드롭아웃에 대해서 말해보세요

- 차원축소 방법
    - pca말했는데 variation이 크면 좋은거냐
